front/src/applications/editor/types.ts

```diff
@@ -23,6 +24,12 @@ export type MapState = {
   viewport: ViewState;
 };
 
+// EDITOAST OPERATIONS
+
+export type CreateOperation = Operation & { operation_type: 'CREATE' };
```

Comment:
When narrowing the type like that, if I hover on `CreateOperation`, I have that : 

![Capture d’écran 2024-06-14 à 16 20 16](https://github.com/OpenRailAssociation/osrd/assets/113104137/cde69429-05a6-4977-a6ff-f3a3918fb010)

We can see that `operation_type: 'CREATE'` appears two times. It doesn't result to any error as we can't assign two times the same property in the object, but a moree elegant way would be to define it like that : 

```suggestion
export type CreateOperation = Extract<Operation, { operation_type: 'CREATE' }>;
```